packages/puppeteer-core/src/cdp/BrowserContext.ts

```diff
@@ -100,6 +101,36 @@ export class CdpBrowserContext extends BrowserContext {
     await this.#browser._disposeContext(this.#id);
   }
 
+  override async cookies(): Promise<Cookie[]> {
+    const {cookies} = await this.#connection.send('Storage.getCookies', {
+      browserContextId: this.#id || undefined,
+    });
+    return cookies.map(cookie => {
+      return {
+        ...cookie,
+        partitionKey: cookie.partitionKey
+          ? cookie.partitionKey.topLevelSite
+          : undefined,
+      };
+    });
+  }
+
+  override async setCookie(...cookies: Cookie[]): Promise<void> {
+    return await this.#connection.send('Storage.setCookies', {
+      cookies: cookies.map(cookie => {
+        return {
+          ...cookie,
+          partitionKey: cookie.partitionKey
+            ? {
+                topLevelSite: cookie.partitionKey,
+                hasCrossSiteAncestor: false,
```

Comment:
hasCrossSiteAncestor can be true or false, this one of the partition keys. The old Cookie object from the page does not support it but we want to support all partition keys (also the ones supported by Firefox over WebDriver BiDi) with the new API.

Reply:
Yeah, I updated the pages API as well, so it could simply be deleted in the future